regression/strings-smoke-tests/java_format/test.java

```diff
@@ -0,0 +1,10 @@
+public class test
+{
+    public static String main()
+    {
+        String s = String.format("foo %s", "bar");
+        assert(s.equals("foo bar"));
+        assert(!s.equals("foo bar"));
+        return s;
+    }
+}
```

Comment:
I added a test where the second argument is not constant and this works fine
```
public static String main(String str)
    {
        String s = String.format("foo %s", str);
        assert(s.equals("foo ".concat(str)));
        assert(!s.equals("foo ".concat(str)));
        return s;
    }
```
When the first argument of format (the format specifier) is not constant then the result should be non-deterministic I can add an example of that (marked as KNOWNBUG) and some others with indexes, and more than 10 arguments for example.